docs/testing-with-django.md

```diff
@@ -1,8 +1,237 @@
 # Backend Django tests
 
-These live in `zerver/tests/tests.py` and `zerver/tests/test_*.py`. Run
-them with `tools/test-backend`.
+## Overview
 
-See [this Github issue](https://github.com/zulip/zulip/issues/879) for ideas
-on improving this documentation.
+Zulip uses the Django framework for its Python back end.  We
+use the testing framework from
+[django.test](https://docs.djangoproject.com/en/1.10/topics/testing/)
+to test our code.  We have over a thousand automated tests that verify that
+our backend works as expected. 
+
+All changes to the Zulip backend code should be supported by tests.  We
+enforce our testing culture during code review, and we also use
+coverage tools to measure how well we test our code.  We mostly use
+tests to prevent regressions in our code, but the tests can have
+ancillary benefits such as documenting interfaces and influencing
+the design of our software.
+
+If you have worked on other Django projects that use unit testing, you
+will probably find familiar patterns in Zulip's code.  This document
+describes how to write tests for the Zulip back end, with a particular
+emphasis on areas where we have either wrapped Django's test framework
+or just done things that are kind of unique in Zulip.
+
+## Running tests
+
+Our tests live in `zerver/tests/`. You can run them with
+`./tools/test-backend`. It generally takes about a minute to run
+the entire test suite.  When you are in iterative mode, you
+can run individual tests or individual modules, following the
+dotted.test.name convention below:
+
+    cd /srv/zulip
+    ./tools/test-backend zerver.tests.tests.WorkerTest
```

Comment:
This totally belongs as a separate issue (https://github.com/zulip/zulip/issues/1670) but it'd be nice if you could just pass WorkerTest here :)